src/kubeobjects/daemonset.go

```diff
@@ -23,6 +24,17 @@ func CreateOrUpdateDaemonSet(c client.Client, logger logr.Logger, desiredDs *app
 		return false, nil
 	}
 
+	if !reflect.DeepEqual(currentDs.Spec.Selector.MatchLabels, desiredDs.Spec.Selector.MatchLabels) {
+		logger.Info("immutable section changed on daemonset, deleting and recreating", "name", desiredDs.Name)
+		err = c.Delete(context.TODO(), currentDs)
+		if err != nil {
+			return false, err
+		}
+		logger.Info("deleted daemonset")
+		logger.Info("recreating daemonset", "name", desiredDs.Name)
+		return true, c.Create(context.TODO(), desiredDs)
```

Comment:
refactor into function to keep the parent function more concise and readable.
also, `c` should be changed to a name